arguments.go

```diff
@@ -27,15 +27,27 @@ import (
 	"io/ioutil"
 )
 
+type ArgReader interface {
```

Comment:
Can you add comments to the exported interfaces here and below. (`go vet` / `golint` should complain about this too, suggest setting those up to run on save)